src/components/TOCViewer/index.jsx

```diff
@@ -1,32 +1,147 @@
+    this.setExpanded = this.setExpanded.bind(this);
+
+    this.state = {
+      expanded: this.props.node.id === this.props.currentPageId,
+    };
+  }
+  componentDidUpdate(prevProps) {
+    const isBeingViewed = this.props.node.id === this.props.currentPageId;
+    const pageHasChanged = this.props.currentPageId !== prevProps.currentPageId;
+
+    if (isBeingViewed && pageHasChanged && this.props.expandParent) {
+      this.props.expandParent();
+    }
+  }
+
+  setExpanded(event) {
+    this.setState({
+      expanded: event ? !this.state.expanded : true,
+    });
+    if (event === undefined && this.props.expandParent) {
+      this.props.expandParent();
+    }
+  }
+
+  render() {
+    return (
+      <li
+        key={this.props.node.id}
+        className={
+          classNames({
+            collapsed: this.props.node.children && !this.state.expanded,
+          })
+        }
+      >
+        <span
+          className={
+            classNames({
+              highlight: this.props.node.id === this.props.currentPageId,
+            })
+          }
+        >
+          {
+            this.props.node.children ?
+              <Button
+                aria-expanded={this.state.expanded}
+                label={
+                  <Icon
+                    className={
+                      classNames({
+                        fa: true,
+                        'fa-chevron-down': this.state.expanded,
+                        'fa-chevron-right': !this.state.expanded,
+                      }).split(' ')
+                    }
+                  />
+                }
+                onClick={this.setExpanded}
+              /> :
+              <span className="bullet">&bull;</span>
+          }
+          <Link to={`/${this.props.journalId}/pages/${this.props.node.id}`}>{this.props.node.title}</Link>
+        </span>
+        {
+          this.props.node.children &&
+          <ul>
+            <TreeList
+              structure={this.props.node.children}
+              journalId={this.props.journalId}
+              currentPageId={this.props.currentPageId}
+              expandParent={this.setExpanded}
+            />
+          </ul>
+        }
+      </li>
+    );
+  }
+}
+
+const TreeList = props => (
   props.structure.map(node => (
-    <li>
-      <Link to={`/${props.journalId}/pages/${node.id}`}>{node.title}</Link>
-      <ul>
-        {node.children && <TreeViewer structure={node.children} journalId={props.journalId} />}
-      </ul>
-    </li>
+    <TreeViewer
+      key={node.id}
+      node={node}
+      journalId={props.journalId}
+      currentPageId={props.currentPageId}
+      expandParent={props.expandParent}
+    />
   ))
 );
 
 const TOCViewer = props => (
   // If the title is an empty string do not display one
-  <div className="toc-border">
-    {props.journal.title.trim() && <div className="journal-title">{props.journal.title}</div>}
+  <div className="toc">
+    <div className="toc-title">TABLE OF CONTENTS</div>
```

Comment:
general question here about labels: what is our strategy for i18n of strings? In django templates we have the _() get text method, guessing we need equivalent for React.

Reply:
I know it's been discussed, but I don't know where it stands. I can ask in FedX on Thursday and if we have a plan in place, I can add a story to get i18n working in the app on our backlog.